korma/src/commonMain/kotlin/com/soywiz/korma/geom/Vector3D.kt

```diff
@@ -5,95 +5,190 @@ import com.soywiz.korma.interpolation.interpolate
 import com.soywiz.korma.math.*
 import kotlin.math.*
 
-class Vector3D {
-    val data = floatArrayOf(0f, 0f, 0f, 1f)
+inline fun vectorLength(x: Float, y: Float, z: Float) = sqrt(vectorLengthSquared(x, y, z))
+inline fun vectorLengthSquared(x: Float, y: Float, z: Float) = (x * x) + (y * y) + (z * z)
 
+class Vector3D constructor(
+    val data: FloatArray
+) {
     var x: Float get() = data[0]; set(value) = run { data[0] = value }
     var y: Float get() = data[1]; set(value) = run { data[1] = value }
     var z: Float get() = data[2]; set(value) = run { data[2] = value }
-    var w: Float get() = data[3]; set(value) = run { data[3] = value }
```

Comment:
This was not voted. Removing `w`, would require providing Vector4 and Vector3.

The idea is that Vector3D both supports Vector3 and Vector4 with a limited API. As commented, mimics the AS3 API:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/geom/Vector3D.html

If you plan to propose this, please keep the Vector3D maybe as deprecated, and introduce two new classes: Vector3 and Vector4. Or at least just Vector3, but keeping Vector3D.